en/components/grid.md

```diff
@@ -543,6 +543,144 @@ public ngOnInit() {
+    </igx-column>
+</igx-grid>
+<button (click)="enableSummary()"> Enable Summary</button>
+```
+```typescript
+public enableSummary() {
+    this.grid1.enableSummaries([{fieldName: "ReorderLevel", customSummary: this.mySummary},
+    {fieldName: "ProductID"}]);
+  }
+```
+
+If these functions do not fulfill your requirements you can provide a custom summary for the specific columns. In order to achieve this you have to override one of the base classes `IgxSummaryOperand`, `IgxNumberSummaryOperand`, `IgxDateSummaryOperand` according to the column data type and your needs. In this way you can redifine the existing or you can add new functions. `IgxSummaryOperand` class provide default implementation only for `count` method. `IgxNumberSummaryOperand` extends `IgxSummaryOperand` and provide implementation for methods: `min`, `max`, `sum` and `average`. `IgxDateSummaryOperand` extends `IgxSummaryOperand` and additionally gives you: `earliest` and `latest`.
+
+```typescript
+import { IgxSummaryResult, IgxSummaryOperand, IgxNumberSummaryOperand, IgxDateSummaryOperand } from 'igniteui-angular/grid/grid-summary';
+
+class MySummary extends IgxNumberSummaryOperand {
+
+  constructor() {
+    super();
+  }
+  operate(data?: any[]): IgxSummaryResult[] {
+    const result = super.operate(data);
+    result.push({
+      key: 'test',
+      label: 'Test',
+      summaryResult: data.filter(rec => rec > 10 && rec < 30).length
+    });
+
+    return result;
+  }
+}
+```
+
+In the code below you can see that method **operate** returns a list of **IgxSummaryResult**, which is an interface.
+```typescript
+interface IgxSummaryResult {
+    key: string;
+    label: string;
+    summaryResult: any;
+}
+```
+And now let's add our custom summary to the column `UnitsInStock`. We will achieve that as we set the property `summaries` to the class, we created below.
+```typescript
+<igx-grid #grid1 [data]="data" [autoGenerate]="false" height="800px" width="800px" (onColumnInit)="initColunm($event)" >
+    <igx-column field="ProductID" width="200px"  [sortable]="true">
+    </igx-column>
+    <igx-column field="ProductName" width="200px" [sortable]="true" [hasSummary]="true">
+    </igx-column>
+    <igx-column field="UnitsInStock" width="200px" [dataType]="'number'" [hasSummary]="true" [summaries]="mySummary" [sortable]="true">
+    </igx-column>
+    <igx-column field="ReorderLevel" width="200px" [editable]="true" [dataType]="'number'" [hasSummary]="true">
+    </igx-column>
+</igx-grid>
+```
+
+```typescript
+...
+export class GridComponent implements OnInit {
+
+  mySummary = MySummary;
+
+    ....
+}
+```
+
+It is good to keep in mind that in order to have a better performance **igx-grid** cache all summaries and recalculate them if you perform CRUD operation over you data. But if your data source is modified outside the **igx-grid**, you need to explicitly force the **igx-grid** to recalculate your summaries by invoking the method `clearSummaryCache()`.
```

Comment:
in order to improve performance in the **igx-grid**, you can cache all summaries and recalculate them if you perform CRUD operations over your data.